content/shmem_finalize.tex

```diff
@@ -24,7 +24,7 @@
     releases resources used by the \openshmem library. This collective
     operation requires all \acp{PE} to participate in the call. There is an
     implicit global barrier in \FUNC{shmem\_finalize} so that pending
-    communications are completed, and no resources can be released until all
+    communications \newtext{by all threads of the \ac{PE}} are completed, and no resources can be released until all
```

Comment:
> this is to emphasize that the User should complete all communications on all threads.

@manjugv Except that the text specifies an "_implicit_ global barrier", which does just the opposite; it emphasizes to the user that the library will take care of completing communication.